tensorboard/webapp/metrics/store/metrics_reducers.ts

```diff
@@ -619,6 +619,20 @@ const reducer = createReducer(
       cardStateMap: nextcardStateMap,
     };
   }),
+  on(actions.metricsCardFullSizeToggled, (state, {cardId}) => {
+    const nextcardStateMap = {...state.cardStateMap};
+    nextcardStateMap[cardId] = {
+      ...nextcardStateMap[cardId],
+      // When the action is fired and card id does not exist on cardStateMap,
+      // this means the full size icon is clicked.
+      fullSize: !nextcardStateMap[cardId]?.fullSize,
+    };
```

Comment:
When the card is set to full size, the table should be expanded. Please add tests for this new reducer
```suggestion
    const nextcardStateMap = {...state.cardStateMap};
    const nextShowFullSize = !nextcardStateMap[cardId]?.fullSize;
    const nextTableExpanded = nextShowFullSize || Boolean(nextcardStateMap[cardId]?.tableExpanded);
    nextcardStateMap[cardId] = {
      ...nextcardStateMap[cardId],
      // When the action is fired and card id does not exist on cardStateMap,
      // this means the full size icon is clicked.
      fullSize: nextShowFullSize,
      tableExpanded: nextTableExpanded,
    };
```